builtin/logical/cassandra/util.go

```diff
@@ -64,18 +62,25 @@ func createSession(cfg *sessionConfig, s logical.Storage) (*gocql.Session, error
 			}
 			if len(cfg.IssuingCA) > 0 {
 				certBundle.IssuingCA = cfg.IssuingCA
-				tlsConfig.InsecureSkipVerify = false
 			}
 
 			parsedCertBundle, err := certBundle.ToParsedCertBundle()
 			if err != nil {
-				return nil, fmt.Errorf("Error parsing certificate bundle: %s", err)
+				return nil, fmt.Errorf("failed to parse certificate bundle: %s", err)
 			}
 
 			tlsConfig, err = parsedCertBundle.GetTLSConfig(certutil.TLSClient)
-			if err != nil {
-				return nil, fmt.Errorf("Error getting TLS configuration: %s", err)
+			if err != nil || tlsConfig == nil {
+				return nil, fmt.Errorf("failed to get TLS configuration: tlsConfig:%#v err:%v", tlsConfig, err)
 			}
+			tlsConfig.InsecureSkipVerify = cfg.InsecureTLS
+
+			var ok bool
+			tlsConfig.MinVersion, ok = tlsutil.TLSLookup[cfg.TLSMinVersion]
```

Comment:
This uses the `parsedCertBundle.GetTLSConfig` call, but below you modified that to return a min version of 1.2. For upgrade reasons, if the value isn't set in the backend config, I'd reset that value to `0`.